streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -60,15 +74,111 @@ public void addChild(ProcessorNode<?, ?> child) {
         children.add(child);
     }
 
-    public void init(ProcessorContext context) {
+    public void init(ProcessorContext context, StreamsMetrics metrics) {
```

Comment:
There seem to be two ways to make this happen

a. for the ProcessorNode to accept Metrics in the constructor, the ProcessorNodeFactory would have to have a pointer to metrics, which would imply that addProcessor in TopologyBuilder would need to have that, and since metrics is internal  to KStreams, users of TopologyBuilder do not have a pointer to metrics.
Alternatively 
b. the build method of ProcessorNodeFactory can take metrics in addition to applicationId and then a ProcessorNode can be constructed by passing Metrics in the ProcessorNode constructor as opposed to piggy backing on the StreamsMetrics which the current review request shows (which is not uniform with StreamThread...)

b,  would look like this 

    @Override
        public ProcessorNode build(String applicationId, Metrics metrics) {
            return new ProcessorNode(name, supplier.get(), stateStoreNames, metrics);
        }